plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/ElasticsearchAwsModule.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch.aws;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.google.inject.Binder;
+import com.google.inject.Singleton;
+import com.google.inject.multibindings.ProvidesIntoOptional;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.plugin.elasticsearch.client.ForElasticsearchSigner;
+import org.apache.http.HttpRequestInterceptor;
+
+import static com.google.inject.multibindings.ProvidesIntoOptional.Type.DEFAULT;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class ElasticsearchAwsModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        configBinder(binder).bindConfig(AwsSecurityConfig.class);
+    }
+
+    /**
+     * Provides a default Optional {@link HttpRequestInterceptor} to sign requests to an Elasticsearch cluster in AWS.
+     * If set, uses the configured aws access and security keys in {@link AwsSecurityConfig}, otherwise defaults to
+     * the {@link DefaultAWSCredentialsProviderChain}
+     *
+     * @param awsConfig aws security config
+     * @return AWS Request Signer
+     */
+    @ProvidesIntoOptional(DEFAULT)
```

Comment:
Instead of using optional binder here, can we have `ElasticsearchAwsModule` to be a default extension in `ElasticsearchConnectorFactory`?

Reply:
I applied having the `ElasticsearchAwsModule` as a default extension, but we still want this binding to be `Optional` because it will only have a value if `elasticsearch.security` is set to `AWS`. 

This logic was already existing in `ElasticsearchClient`, so it knows when to sign requests with AWS credentials.